plugins/exec/resources.mli

```diff
@@ -0,0 +1,24 @@
+(* FreeSpec
+ * Copyright (C) 2018–2020 ANSSI
+ *
+ * Contributors:
+ * 2020 Thomas Letan <thomas.letan@ssi.gouv.fr>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *)
+
+val insert : 'a -> int
```

Comment:
Is it important to make public the fact that resources handlers are implemented with integers?

Otherwise, I would introduce an abstract type here.

Reply:
Considering they are injecting within Coq AST,one possible approach could be to return a `Constr.t`.

Reply:
I did just that in an additional commit.

Reply:
Oh I see! That's clearly the right way to go.